integration_tests/src/examples.rs

```diff
@@ -135,8 +145,102 @@ pub fn start_ffplay(ip: &str, video_port: u16, audio_port: Option<u16>) -> Resul
     Ok(())
 }
 
-#[allow(dead_code)]
-pub fn stream_video(ip: &str, port: u16, path: PathBuf) -> Result<()> {
+// plays H264 encoded video and Opus encoded audio using gstreamer. If no port given, returns Error
+pub fn start_gstplay(ip: &str, port: u16, video: bool, audio: bool) -> Result<()> {
+    let mut gst_output_command = [
+        "gst-launch-1.0 -v ",
+        "rtpptdemux name=demux ",
+        &format!("tcpclientsrc host={} port={} ! \"application/x-rtp-stream\" ! rtpstreamdepay ! queue ! demux. ", ip, port)
+        ].concat();
+
+    if video {
+        gst_output_command.push_str("demux.src_96 ! \"application/x-rtp,media=video,clock-rate=90000,encoding-name=H264\" ! queue ! rtph264depay ! decodebin ! videoconvert ! autovideosink ");
+    }
+    if audio {
+        gst_output_command.push_str("demux.src_97 ! \"application/x-rtp,media=audio,clock-rate=48000,encoding-name=OPUS\" ! queue ! rtpopusdepay ! decodebin ! audioconvert ! autoaudiosink ");
+    }
+
+    Command::new("bash")
+        .arg("-c")
+        .arg(gst_output_command)
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn()?;
+    thread::sleep(Duration::from_secs(2));
+
+    Ok(())
+}
+
+// GStreamer will stream with H264 video encoding and Opus audio encoding if video and audio ports will be given. Will return Error otherwise.
```

Comment:
All those errors are pretty obvious, you don't have to mention them.
```suggestion
```

Usually in Rust, you would define such contracts using a type system (3 variant enum here + typed return error), but here they're so simple and internal, that it's not worth it.